### PR TITLE
Remove split subdirectories as part of #372 modularization

### PR DESCRIPTION
**Pull request by rib**
This PR removes `ndk-glue`, `ndk-macro`, `cargo-apk`, `ndk-build`, `ndk-examples` and `ndk-context` from this repo as part of the modularization effort described in issue #372

- `ndk-glue` and `ndk-macro` have been moved to https://github.com/rust-mobile/ndk-glue
- `cargo-apk`, `ndk-build` and `ndk-examples` have been moved to https://github.com/rust-mobile/cargo-apk
- `ndk-context` has moved to https://github.com/rust-mobile/ndk-context

Supersedes: #373

### CODE REVIEW

**Review by dvc94ch — APPROVED** [review]

**Comment by rib** [comment]
@MarijnS95 it would be good if you could also ACK/review this PR if you get a moment

**Comment by rib** [comment]
Just a quick ping @MarijnS95 since I see you're looking at the standalone `cargo-apk` repo now.

I'd like to use the merging of this PR as the final commitment that the standalone repos are acceptable so if you've had a chance to review the state of the new repo it would be good if you can ACK here that you're happy for this to be merged.

I take it @dvc94ch is happy with the ndk-context repo, and hopefully you are both ok with the ndk-glue repo too.

My stance is that until we merge this then we can still potentially re filter the standalone repos if we absolutely need to so we probably shouldn't go too far ahead with using the new repos before agreeing that we're happy with them?

I think if we can get an OK from you @MarijnS95 then we can probably just go ahead and merge this.

**Comment by MarijnS95** [comment]
> Just a quick ping @MarijnS95 since I see you're looking at the standalone cargo-apk repo now.

It's Monday now so I'm _slowly_ going though all the things ;)

File: .github/workflows/rust.yml
Comment (MarijnS95) [inline .github/workflows/rust.yml:79]: /me pukes.

File: .github/workflows/rust.yml
Comment (MarijnS95) [inline .github/workflows/rust.yml:60]: I saw you created an issue about the `abi` mess, hope it'll be resolved soon.

File: .github/workflows/rust.yml
Comment (MarijnS95) [inline .github/workflows/rust.yml:33]: Can you clean this up in the inverse way in the `cargo-apk` dir, since it specifically does not have to run on the target arch anymore, but on a (set of) host(s)?

OTOH that host might as well be an Android phone; linting for all the triples is fine by me but the comment is dated.

**Comment by MarijnS95** [comment]
This PR doesn't seem to update the root `README.md`, nor has a new root readme been added (or symlinked) in at least `cargo-apk`. Can you look into that? I'm not sure what we should make it look like, the matrix was always relevant and useful to easily click through, adding repository links and a small "use with this crate to achieve X" description might be nice too (since most crates are rather standalone and compatible with alternative systems now).

**Comment by dvc94ch** [comment]
When did we decide to comply with the ridiculous woke agenda and rename all branches to main? I don't live in an oppressive white patriarchy where people are so traumatized by their ancestors enslavement that happened to people of all races that they have PTSD when they see the word master, despite it being derived from the older term master copy, not slave master. I'm also not interested in appeasing or condoning this sick world view in any way

File: .github/workflows/rust.yml
Comment (rib) [reply .github/workflows/rust.yml:60]: Yeah, it's a bit of a weird quirk with cargo ndk

**Comment by rib** [comment]
> This PR doesn't seem to update the root `README.md`, nor has a new root readme been added (or symlinked) in at least `cargo-apk`. Can you look into that? I'm not sure what we should make it look like, the matrix was always relevant and useful to easily click through, adding repository links and a small "use with this crate to achieve X" description might be nice too (since most crates are rather standalone and compatible with alternative systems now).

Yeah, this is intentional and noted in #372. The intention is to follow up with README changes seperately and not hold up the initial separation (since I guess there will be more discussion / iteration around readme changes)

**Comment by rib** [comment]
> When did we decide to comply with the ridiculous woke agenda and rename all branches to main? I don't live in an oppressive white patriarchy where people are so traumatized by their ancestors enslavement that happened to people of all races that they have PTSD when they see the word master, despite it being derived from the older term master copy, not slave master. I'm also not interested in appeasing or condoning this sick world view in any way

I'm not sure how to respond to this. This is far too heated, and politically charged, for a GitHub PR discussion.

`main` is a conventional name for the main branch that's also the default recommended by GitHub. I copy and pasted the commands from GitHub when pushing the standalone repos.

File: .github/workflows/rust.yml
Comment (MarijnS95) [reply .github/workflows/rust.yml:79]: As long as it works... The CI is green, so it surprisingly does?

**Comment by dvc94ch** [comment]
> main is a conventional name for the main branch that's also the default recommended by GitHub. I copy and pasted the commands from GitHub when pushing the standalone repos.

the traditional name is master, and it bothered no one until recently. it clearly is a political statement, otherwise no one would have bothered to change it. it requires accepting the premises of a radical ideology that is less than a decade old.

- [0] https://en.wikipedia.org/wiki/Demoralization_(warfare)

**Comment by rib** [comment]
Okey, hopefully we can move on from that.

Considering @MarijnS95 has gone ahead and spun a release out of the standalone `cargo-apk` repo, including the addition of tags I think that effectively commits us to the standalone `cargo-apk` repo at this point (any rebasing/re-filtering would change commit hashes) and e.g. from this comment from @MarijnS95 https://github.com/rust-mobile/cargo-apk/pull/4#issuecomment-1347510767 he's ok with moving ahead based on PRs/issues to address any feedback with the commits I made (such as for the porting of examples that @MarijnS95 had some comments on already).

So I think this is ready to merge a this point 🤞

**Review by MarijnS95 — APPROVED** [review]
Fine by me if you can follow up on the readme changes here and elsewhere. I'll slowly start processing more of #372 and the resulting repos but, as mentioned before, I'm quite occupied unfortunately :sweat_smile: :cry:

File: .github/workflows/rust.yml
Comment (rib) [reply .github/workflows/rust.yml:33]: Right, it'd make sense in `cargo-apk` to lint with host targets, since it's a tool that runs on the host. It should be easy to follow up on iterating / cleaning up the CI in any of the standalone repos.

My first changes were just focused on making minimal CI updates that could ensure everything was building with green lights in the standalone repos.

Comment (MarijnS95) [reply .github/workflows/rust.yml:33]: Mostly the comment that's off now, I'm fine having "android" itself as a host if someone wants to build APKs from Android for Android (Termux anyone?).

**Comment by dvc94ch** [comment]
> And I really don't care about main vs master and won't waste a second discussing/thinking about it.

pretty sure writing that sentence took at least a second.

on another note if you care about ios, I'm making some great progress on using xbuild to build/upload apps to the appstore from linux. I think that will be quite an exciting feature once it lands

**Comment by rib** [comment]
I think this is ready to be merged, yeah. @MarijnS95 are you waiting for anything, or maybe just waiting for someone else to merge? I feel like it's up to you to merge as the current maintainer here @MarijnS95.

regarding communicating about rust-mobile I've at least just created an admins team for the organization to be able to use the github discussion boards between the organisation owners.

regarding android_logger - we should certainly extend an invitation for them to put the repo under the rust-mobile org if they want. It looks like [tyranron](https://github.com/tyranron) is happy to pick up maintenance of the repo and could be set as the owner for the repo along with [Nercury](https://github.com/Nercury).

**Comment by dvc94ch** [comment]
well, since they found new maintainers there's probably no need

**Comment by rib** [comment]
The main benefit would probably just be that rust-mobile can serve as a place to collate utility crates that facilitate mobile development with Rust (to help with discoverability) but also for smaller projects like android_logger there's probably a higher chance of maintainers moving on and having them under one umbrella may help with finding someone willing to help with maintenance if wanted in the future.

**Comment by MarijnS95** [comment]
Would be great to have them under the unified umbrella, if only for visibility/collation and to decrease bus factor as mentioned elsewhere :)

> I think this is ready to be merged, yeah. @MarijnS95 are you waiting for anything, or maybe just waiting for someone else to merge? I feel like it's up to you to merge as the current maintainer here @MarijnS95.

I think this PR is ready to go, just curious where we are for the other repos WRT readme and last-minute comments I had (for example to fix the release workflow) - have those been addressed?

**Comment by rib** [comment]
Whoop, thanks for merging @MarijnS95 !

> I think this PR is ready to go, just curious where we are for the other repos WRT readme and last-minute comments I had (for example to fix the release workflow) - have those been addressed?

Some of this is tracked as remaining tasks under https://github.com/rust-mobile/ndk/issues/372 or issues that I think you opened, or hopefully we'll remember some of them in the short term (or I'll also get around to creating more tracking issues too). I was travelling a bit last week so I didn't look at any of those things yet but I was thinking I'd look at opening a PR soon for adding a short-term notice about where some of the subdirectories have moved (before looking at more README updates).

**Comment by MarijnS95** [comment]
Heh hope you can still recover those GitHub notifications before force-pushing, I mostly replied to pushed commits on the repo. It's trivial things but they need to happen nevertheless, otherwise I have to not forget them before entering the release cycle :)

**Pull request merged**
